### PR TITLE
Add breakpoints on studio.code.org signed out header

### PR DESCRIPTION
**Pull request by kelbyhawn**
Adds the same breakpoint UI on logged out pages on https://studio.code.org that are on https://code.org (see related PR: https://github.com/code-dot-org/code-dot-org/pull/59016).
- Adds a `SHOW_SMALL_DESKTOP` constant and `show-small-desktop` and `hide-small-desktop` classes to show and hide elements between the small desktop breakpoint (`1268x1024px`).
- Refactored some of the existing hamburger css to be shared across the sites.

## Links
Jira ticket: [ACQ-1975](https://codedotorg.atlassian.net/browse/ACQ-1975)

## Testing story
Tested locally and updated unit test

----

## Signed out (code.org)

https://github.com/code-dot-org/code-dot-org/assets/9256643/6c16c096-73b5-43b1-abcf-214304b05277

## Signed out (studio.code.org)

https://github.com/code-dot-org/code-dot-org/assets/9256643/8c20ca9e-4990-477f-a134-068faa17ebf1

## Signed out level (studio.code.org) - _No change_

https://github.com/code-dot-org/code-dot-org/assets/9256643/2e844739-121f-4bc4-96a3-1d29c45023c1

## Signed in level (studio.code.org) - _No change_

https://github.com/code-dot-org/code-dot-org/assets/9256643/f6e73e1d-1ea1-41f4-9364-2f88f49cd84d

## Signed in Student (studio.code.org) - _No change_

https://github.com/code-dot-org/code-dot-org/assets/9256643/3096b332-912f-4503-a297-c36e76e69415

### CODE REVIEW

File: pegasus/sites.v3/code.org/styles_min/040-page.css
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/styles_min/040-page.css:1]: Was able to move all of these styles into the shared `hamburger.scss` file 🙌

File: shared/haml/help_button.haml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
   help_items = HelpHeader.get_help_contents(options)
 
-.help_button{class: "hide-mobile", id: "help-button", tabindex: "0", 'aria-label': I18n.t('header_screen_reader_help')}
+.help_button{class: user_type ? "hide-mobile" : "show-small-desktop", id: "help-button", tabindex: "0", 'aria-label': I18n.t('header_screen_reader_help')}
```
Comment (kelbyhawn) [inline shared/haml/help_button.haml:8]: This applies the `show-small-desktop` only if a user is logged out.

File: shared/css/user-menu.scss [hunk 2]
```diff
@@ -193,4 +200,16 @@
       padding-left: 10px;
     }
   }
+
+  @media screen and (max-width: 1023px) {
```
Comment (kelbyhawn) [inline shared/css/user-menu.scss:204]: This is controlling the Create/New Project menu button.

File: shared/css/user-menu.scss [hunk 3]
```diff
@@ -116,6 +116,13 @@
     display: block;
   }
 
+  @media (max-width: 1268px) {
```
Comment (kelbyhawn) [inline shared/css/user-menu.scss:119]: This is controlling the Help icon button.

File: dashboard/app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -230,8 +230,9 @@ img.video_thumbnail {
     }
   }
 
-  @media screen and (max-width: 1120px) {
-    .create_menu {
+  @media (min-width: 1024px) and (max-width: 1268px) {
```
Comment (kelbyhawn) [inline dashboard/app/assets/stylesheets/application.scss:233]: This applies to studio.code.org only.

File: pegasus/sites.v3/code.org/styles_min/040-page.css
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/styles_min/040-page.css:1]: The styles removed here have been reworked in the other stylesheets in this PR.

File: shared/css/user-menu.scss (hunk 2 above)
Comment (TurnerRiley) [reply shared/css/user-menu.scss:204]: Thanks for labelling these changes!

**Review by TurnerRiley — APPROVED** [review]

**Pull request merged**